plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -668,8 +692,12 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         if (type == DATE) {
             return WriteMapping.longMapping("Date", dateWriteFunctionUsingLocalDate(getClickHouseServerVersion(session)));
         }
-        if (type == TIMESTAMP_SECONDS) {
-            return WriteMapping.longMapping("DateTime", timestampSecondsWriteFunction(getClickHouseServerVersion(session)));
+        if (type instanceof TimestampType timestampType) {
```

Comment:
Q: we map DateTime64 to TimestampWithTimeZone. But we also map Timestamp to DateTime64 (reverse direction).

Why not same Trino type in both directions?

Reply:
> Timestamp(Trino) -> DateTime64(ClickHouse)

Because Trino Timestamp supports a larger range than either ClickHouse DateTime or DateTime64, and DateTime64 supports a larger range than DateTime, so to support a larger range, I think we should map Trino Timestamp to ClickHouse DateTime64.